### ISSUE TITLE
Deploying server.js on production

### ISSUE THREAD

**Issue by baj84** [twilio/twilio-video-app-react#354]
**Question**
I've used server.js as a wrapper for calling my own API with my own accessToken. It's the API that acts as the token server, but i'm using this proxy so that my API token is kept secret from the frontend.

In my server.js I have:
```
const apiUrl = process.env.TALKED_API_URL;
const apiKey = process.env.TALKED_API_KEY;

app.get('/token', (req, res) => {
  const { identity, roomName } = req.query;

  const data = {
    identity,
    roomName,
    key: apiKey,
  };

  axios
    .post(apiUrl + '/video/token', data)
    .then(result => {
      res.send(result.data);

      console.log(`issued token for ${identity} in room ${roomName}`);
    })
    .catch(error => {
      console.log(`failed to get token for ${identity}`);

      res.status(500);
      res.send();
    });
});
```
The issue is when I deploy this using the command `npm run deploy:twilio-cli -- --override` it's clear that the server.js is running and it's using the serverless version of /token. How can I use my custom code in server.js? Do I need to split it up into serverless functions? OR can i somehow get server.js node server running on Twilio?

**Comment by timmydoza** [comment]
Thanks for the question @baj84!

The `server.js` file that is in this repo is for local development only. It is what is run when you run `npm start`. It is not the code that is deployed when you run `npm run deploy:twilio-cli`. 

The code you are looking for is in our [plugin-rtc](https://github.com/twilio-labs/plugin-rtc) repo. Specifically, [this file](https://github.com/twilio-labs/plugin-rtc/blob/master/src/video-token-server.js).

You can configure the Twilio CLI so that it runs a local version of plugin-rtc. You can then modify the local code for plugin-rtc, so when you run `npm run deploy:twilio-cli`, it will run your modified code. There are some instructions in the [README](https://github.com/twilio-labs/plugin-rtc#install-the-plugin-for-local-development).

Additionally, we added a new feature to the plugin-rtc repo that you may like! First, make sure you have the newest version of the plugin installed with `twilio plugins:update`. Once this is finished, deploy a new video app. Once the deployment is complete, you'll see a message like this in your console:

```
Edit your token server at: https://www.twilio.com/console/functions/editor/ZSxxxxxxxxxxxxxxxx/environment/ZExxxxxxxxxxxxxxxx/function/ZHxxxxxxxxxxxxxxxx
```

Just click on that link and it will take you to our Functions Editor inside the Twilio Console. There you'll be able to edit the live token server right inside your browser.

Let me know how this works out for you! And please let me know if you have any further questions.

**Comment by baj84** [comment]
Thanks @timmydoza for your help. I checked out plugin-rtc into my directory and tried to link it, but I got the following error:
```
plugin: @twilio-labs/plugin-rtc
root: /Users/ben/code/talked/talked-video/plugin-rtc
See more details with DEBUG=*
(node:63626) Error Plugin: @twilio-labs/plugin-rtc: could not find package.json with {
  type: 'dev',
  root: '/Users/ben/code/talked/talked-video/plugin-rtc',
  name: '@oclif/plugin-help'
}
```

Also, once this is working and I modify the video-token-server.js with my own code will it automatically get added as a function when I do a deployment?

And one final question :) I don't have a need for this passcode validation - am I safe to remove it?
My authentication logic is essentially a token in the URL string (each party is given a unique token), that is then sent to my own API layer which validates the users token against the DB and if valid will then create the room using Twilio SID + Secret - Is this an OK way to do things?
